hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/SchedulePreTransactionHandlerImpl.java

```diff
@@ -75,24 +78,54 @@ public ScheduleTransactionMetadata preHandleCreateSchedule(
         // to see
         // if provided payer is same as payer in the inner transaction.
 
-        final var innerMeta = preHandleInnerTxn(scheduledTxn, payerForNested, dispatcher);
+        final var innerMeta = preHandleScheduledTxn(scheduledTxn, payerForNested, dispatcher);
         meta.scheduledMeta(innerMeta);
         return meta.build();
     }
 
-    private TransactionMetadata preHandleInnerTxn(
+    @Override
+    public ScheduleTransactionMetadata preHandleSignSchedule(
+            final TransactionBody txn,
+            final AccountID payer,
+            final PreHandleDispatcher dispatcher) {
+        final var op = txn.getScheduleSign();
+        final var id = op.getScheduleID();
+
+        final var scheduleLookupResult = scheduleStore.get(id);
+        if (scheduleLookupResult.isEmpty()) {
+            return new InvalidTransactionMetadata(txn, payer, INVALID_SCHEDULE_ID);
```

Comment:
What is the expected behavior when this happens? Right now the workflow will move on, validate the signatures and publish the transaction. Is that intended?

Reply:
Yes, as discussed in meeting, it is intended

Reply:
But as discussed in the meeting, we really should throw an exception rather than returning an InvalidTransactionMetadata.

The workflow should have a try/catch. Try calling preHandle, and catch both`PreCheckException`s and `Throwable`s. In all cases, the preHandles will throw on invalid, and workflow will catch and set the tx metadata (so InvalidTransactionMetadata would be an implementation detail from the app, and not something the services would see or know about).